Adds missing keys from i18n tasks output

filipefurtad0 opened the pull request:
#### What? Why?

Related to  #11938.

<!-- Explain why this change is needed and the solution you propose.
     Provide context for others to understand it. -->

The output from `i18n-tasks missing` has been added to this PR - it will be deleted once all the entries have been verified. The idea is to check all of them and either ignore (if the key is found on en.yml - this seems to happen for dynamic keys) or add it to en.yml it it is indeed verified to be missing.

#### What should we test?
<!-- List which features should be tested and how.
     This can be similar to the Steps to Reproduce in the issue.
     Also think of other parts of the app which could be affected
     by your change. -->

A list of affected pages will be updated here, according to progress:
- `/admin/stripe_connect_settings/edit` -> Instance Publishable Key
![image](https://github.com/openfoodfoundation/openfoodnetwork/assets/49817236/d89a411d-ceb9-4d2a-b356-57de2f4893e3)


#### Release notes

<!-- Please select one for your PR and delete the other. -->

Changelog Category (reviewers may add a label for the release notes):

- [x] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [ ] Technical changes only
- [ ] Feature toggled

<!-- Choose a pull request title above which explains your change to a
     a user of the Open Food Network app. -->

The title of the pull request will be included in the release notes.


#### Dependencies
<!-- Does this PR depend on another one?
     Add the link or remove this section. -->



#### Documentation updates
<!-- Are there any wiki pages that need updating after merging this PR?
     List them here or remove this section. -->

filipefurtad0 commented on missing_keys.txt:
```diff
@@ -47,7 +47,7 @@
 |  all   | spree.taxon_edit                                                             | app/views/spree/admin/taxons/edit.html.haml:4                                                      | - missing translation, added key
-|  all   | thank_you_for_your_order                                                     | app/views/spree/orders/show.html.haml:26                                                           |
+|  all   | thank_you_for_your_order                                                     | app/views/spree/orders/show.html.haml:26                                                           | - missing translation, added key
 |  all   | unrecognized_card_type                                                       | app/views/spree/admin/payments/source_forms/_gateway.html.haml:25                                  |
```
Another option could have been to reuse the key `thanks: "Thank you for your business."`.

filipefurtad0 commented on config/locales/en.yml:
```diff
@@ -259,6 +259,10 @@ en:
+  unrecognized_card_type: ""
+  use_new_cc: "Use a new credit card"
+  what_is_this: "What is this?"
+  your_cart_is_empty: "Your cart is empty"
```
I'm not sure these keys are needed... But they are indeed missing.

mkllnk reviewed (approved):
Great pull request! Easy to follow, step by step.

And nice cleanup along the way. :heart_eyes:

filipefurtad0 commented:
I've noticed there were these two files (which I've marked as needing issues to fix missing translations):

- app/views/home/_connect.html.haml
- app/views/home/_learn.html.haml

However, after a second look, I'm convinced they are not in use, so I've removed them still within this PR. Re-requesting review :pray:

rioug reviewed (approved):
Nice one ! Thanks for that :pray: , it seems like an unpleasant task.

mkllnk reviewed (approved):
> I'm convinced they are not in use, so I've removed them still within this PR.

Thank you! I forgot to remove them in a712f254e0424f7756f73f49103e161802e21cd7.

RachL commented:
@filipefurtad0 will this PR close https://github.com/openfoodfoundation/openfoodnetwork/issues/2626 ?

filipefurtad0 commented:
> will this PR close https://github.com/openfoodfoundation/openfoodnetwork/issues/2626 ?

Hey @RachL ,
Not exactly. This PR uses a gem to locally look for missing translations and fixes / removes some unnecessary files.

I'd say the actual PR contributing to #2626 is #11906, as per [this comment](https://github.com/openfoodfoundation/openfoodnetwork/pull/11906#issuecomment-1856833265).

I'm not sure we should keep it open though as we've improved the previous code to detect more types of missing translations, and because we've opted not to implement any of the follow-up suggestions (references, [comment above](https://github.com/openfoodfoundation/openfoodnetwork/pull/11906#issuecomment-1856833265) and [here](https://github.com/openfoodfoundation/openfoodnetwork/pull/11906#discussion_r1432177040)) like:
- integrating a build node to check the build
- and a Bugsnag warning to spot missing translations in production;

So, I'd say yes, let's close!

drummer83 commented:
Hi @filipefurtad0,
Thanks for this PR! I think we are reducing the missing translations bit by bit. :muscle: 

I have found no issues with this PR, only a few questions you might want to have a look at?

1) `/admin/stripe_connect_settings/edit` -> Instance Publishable Key :arrow_right: :heavy_check_mark: I added the key manually for testing:
![image](https://github.com/openfoodfoundation/openfoodnetwork/assets/1790176/620c4285-c256-4ac3-a3cf-eadfa195061a)

2) Verify that translations work as before on all pages, during customer checkout  :arrow_right: :heavy_check_mark: Yes, all working fine.

3) As superadmin, check taxonomy pages, namely editing and creating new taxonomies. Can you find a remove button? Is it rendering correctly?  :arrow_right: I cannot find a remove button which requires this new translation. :question: 

4) As a customer, initiate the checkout process, by adding some items to the cart. Edit the cart. Is the `Continue shopping` button displayed correctly, as before?  :arrow_right: :heavy_check_mark: This one was translated before already:
![image](https://github.com/openfoodfoundation/openfoodnetwork/assets/1790176/da4f69e1-b914-4326-a649-d58c98277c78)
The new one only kicks in when the cart is empty - but I don't know how to get to the edit cart page when the cart is empty.
```
  #cart-container
    - if @order.line_items.empty?
      %div.row
        %p= t(:cart_empty)
        %p= link_to t(:orders_edit_continue), current_shop_products_path, :class => 'button continue'
```
You replaced the old string 'your_cart_is empty' with the new one, but at the same time you added it as missing translation to en.yml line 265. I think it's not needed then anymore? :question: 

5) As admin, edit an order without any payments. Are you able to see any message stating that "the order has no payments"? I believe this was never the case. In summary: going through the side tabs, when editing an order as admin, should display all information correctly. :arrow_right: :heavy_check_mark: I can't make that message appear.

6) Ideally, we would trigger the flash message: "The order could not be updated" but I don't know how to trigger this. Will ask testers and update instructions.  :arrow_right: :question: Testers have no clue. Which commit is dealing with a flash message? 

7) As superadmin, editing states pages should display correctly :arrow_right: :heavy_check_mark: This was a missing translation before. The page still displays correctly and after adding the translation keys manually the translation was applied.
Before:
![image](https://github.com/openfoodfoundation/openfoodnetwork/assets/1790176/95d78e91-806f-4b76-889d-ea3d5d268c23)
After:
![image](https://github.com/openfoodfoundation/openfoodnetwork/assets/1790176/7729ad2f-5a38-4845-bcec-6d26412c7871)

8) As superadmin, under `/admin/contents/edit`, verify that it is possible to enable/disable the `Menu 6 icon name` and `Menu 7 icon name` sections of the homepage. :arrow_right: :heavy_check_mark: Yes, this is working. But which part of the PR is affecting it?

NB: The string 'taxon_edit' is used as page title only on a subpage of a page which uses 'taxonomy_edit' as page title. I think we could combine both of them in the existing 'taxonomy_edit'?

NB2: I have also no idea how the string 'Thank you for your order' can be seen. Do you know?

I will add the feedback-needed label for now until your feedback.

Thanks again! :pray: 
And have a great weekend! :sun_with_face:

filipefurtad0 commented on missing_keys.txt:
```diff
@@ -0,0 +1,55 @@
+|  all   | unrecognized_card_type                                                       | app/views/spree/admin/payments/source_forms/_gateway.html.haml:25                                  |
+|  all   | use_new_cc                                                                   | app/views/spree/admin/payments/source_forms/_gateway.html.haml:10                                  |
+|  all   | what_is_this                                                                 | app/views/spree/admin/payments/source_forms/_gateway.html.haml:42                                  |
+|  all   | your_cart_is_empty                                                           | app/views/spree/orders/edit.html.haml:31                                                           |
```
Ping Konrad :point_up_2:

filipefurtad0 commented:
Hi @drummer83 , thanks for testing this one!

There are cases in which indeed, I don't know how to trigger the respective behavior. Some of them looked like dead code, so I've removed them in this PR. In other cases, I had no basis to assume that, and I felt it is better to add a key, then being to thorough on the grooming, which could actually _create_ missing translations... Adding a key should not have any negative impact on the functionality of the app; worst case scenario, it's adding a key which is never rendered in the frontend. I don't see this as a big risk, so I've added keys on points

- 3
- 6 -> flash message commit is here f13047060d0bc20f2e0bd742ac6a1ca4395c8f8b

On 4:

> You replaced the old string 'your_cart_is empty' with the new one, but at the same time you added it as missing translation to en.yml line 265. I think it's not needed then anymore? 

Good point. I guess this was my initial intention but left the key there, hanging. Done [here](https://github.com/openfoodfoundation/openfoodnetwork/pull/11958/commits/91769574e39e7e31f417e177901900ac7feff2a6). Thanks for spotting it!

> As superadmin, under /admin/contents/edit, verify that it is possible to enable/disable the Menu 6 icon name and Menu 7 icon name sections of the homepage. ➡️ ✔️ Yes, this is working. But which part of the PR is affecting it?

None, the PR should not affect the pages. However, some files were removed, which _could_ affect the page, but are not supposed to. This is the [commit](https://github.com/openfoodfoundation/openfoodnetwork/pull/11958/commits/d2c2e208226ecf3b5ca7fbac0e29552afb0d5099).

> NB: The string 'taxon_edit' is used as page title only on a subpage of a page which uses 'taxonomy_edit' as page title. I think we could combine both of them in the existing 'taxonomy_edit'?

Nice catch! Done on the latest commit https://github.com/openfoodfoundation/openfoodnetwork/pull/11958/commits/bfd4b730f29d52c7e33568a2a4123e7104a21b75.

I think I've addressed everything. There are two new minor commits so I'm moving to code review.

Thanks again @drummer83 :pray:

drummer83 commented:
Thanks, @filipefurtad0, for clarifying and updating the last bits!

I have tested the changes of your latest commits. 
It's all looking good! :tada:
Merging! :rocket:
Thanks again!

dacook commented:
@filipefurtad0 Can you please confirm if this PR contained user-facing changes or not? I quickly scanned and couldn't see any evidence, but could be wrong.

filipefurtad0 commented:
Hey @dacook,
I've marked it as user facing as it introduces some new translation keys, on `en.yml`. I've checked previous releases, I think this is the current process. What do you think?

dacook commented:
Sorry I missed that, yes that sounds good 👍
